openedx/core/djangoapps/discussions/models.py

```diff
@@ -2,19 +2,124 @@
 Provide django models to back the discussions app
 """
 from __future__ import annotations
+import logging
 
 from django.core.exceptions import ValidationError
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
+from django_mysql.models import ListCharField
 from jsonfield import JSONField
 from model_utils.models import TimeStampedModel
 from opaque_keys.edx.django.models import LearningContextKeyField
 from simple_history.models import HistoricalRecords
 
 from lti_consumer.models import LtiConfiguration
+from organizations.models import Organization
 
+from openedx.core.djangoapps.config_model_utils.models import StackedConfigurationModel
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
+log = logging.getLogger(__name__)
+
+
+def get_default_providers():
+    providers = [
+        'cs_comments_service',
+        'lti-base',
+    ]
+    return providers
+
+
+class ProviderFilter(StackedConfigurationModel):
+    allow = ListCharField(
+        base_field=models.CharField(
+            choices=[
+                (provider, provider)
+                for provider in get_default_providers()
+            ],
+            max_length=20,
+        ),
+        blank=True,
+        help_text=_("Comma-separated list of providers to allow, eg: {choices}").format(
+            choices=','.join(get_default_providers()),
+        ),
+        # max_length = (size * (max_length + 1)) # to include the comma
+        max_length=(len(get_default_providers()) * 21),
+        size=len(get_default_providers()),
+        verbose_name=_('Allow List'),
+    )
+    deny = ListCharField(
+        base_field=models.CharField(
+            choices=[
+                (provider, provider)
+                for provider in get_default_providers()
+            ],
+            max_length=20,
+        ),
+        blank=True,
+        help_text=_("Comma-separated list of providers to deny, eg: {choices}").format(
+            choices=','.join(get_default_providers()),
+        ),
+        # max_length = (size * (max_length + 1)) # to include the comma
+        max_length=(len(get_default_providers()) * 21),
```

Comment:
TODO: @xitij2000 @davidjoy I suspect we probably want to actually _hard-code_ this length calculation completely, to an appropriate upper bound.

We don't want a change in config to risk modifying schema and truncating/shortening any column; this value should never automatically decrease.

Reply:
Additionally, the migrations created for this field will be based on the value calculated on the devstack, which could be different from a value for other users (not right now, but eventually). So I think this should just be set to a fixed, reasonably high number.

Reply:
+1 to @xitij2000